pkg/util/validation/stack.go

```diff
@@ -20,16 +20,33 @@ import (
 	"regexp"
 
 	"github.com/pulumi/pulumi/sdk/v3/go/common/apitype"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
 )
 
-// validateStackName checks if s is a valid stack name, otherwise returns a descriptive error.
+// ValidateStackName checks if s is a valid stack name, otherwise returns a descriptive error.
 // This should match the stack naming rules enforced by the Pulumi Service.
-func validateStackName(s string) error {
-	stackNameRE := regexp.MustCompile("^[a-zA-Z0-9-_.]{1,100}$")
-	if stackNameRE.MatchString(s) {
-		return nil
+func ValidateStackName(s string) error {
```

Comment:
The change here only improves the error message. It does not change the validation semantics.